### PR TITLE
feat(dashboard): handle v1 legacy workflows redirect

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What changed? Why was the change needed?

- redirect to legacy editor for v1 workflows
### Screenshots

https://github.com/user-attachments/assets/b0c896fd-524d-4413-b886-acadea72ee8f

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/src/components/workflow-list.tsx [hunk 1]
```diff
@@ -169,6 +169,19 @@ export const WorkflowList = () => {
...
+                          if (workflow.type === WorkflowTypeEnum.REGULAR) {
+                            /**
+                             * React Router doesn't support nested routes, so we need to manually redirect to the legacy route
+                             */
+                            window.location.href = buildRoute(LEGACY_ROUTES.EDIT_WORKFLOW, {
+                              workflowId: workflow._id,
+                            });
+                            return;
+                          }
+
```
Comment (rifont) [inline apps/dashboard/src/components/workflow-list.tsx]: ❓ **question:** Using `onClick` handler results in an inability to "Right-click"->"Open in new tab" (or shortcut `⌘-click`), it's one of my pet peeves when browsing a website, resulting in poor UX.

Can we use the native `href` attribute or React-router `<Link />` component instead? That will preserve the ability to ⌘-click‏

Comment (BiswaViraj) [reply apps/dashboard/src/components/workflow-list.tsx]: Good point.
Updating it to use the link component

File: apps/dashboard/src/components/workflow-list.tsx [hunk 2]
```diff
@@ -154,86 +154,88 @@ export const WorkflowList = () => {
...
+                        {workflow.origin === WorkflowOriginEnum.EXTERNAL && (
+                          <Badge className="rounded-full px-1.5" variant="warning-light">
+                            <BadgeContent variant="warning">
+                              <FaCode className="size-3" />
+                            </BadgeContent>
+                          </Badge>
+                        )}
+                        <Link to={workflowLink} reloadDocument={isV1Workflow}>
+                          <TruncatedText className="cursor-pointer" text={workflow.name} />
+                        </Link>
```
Comment (BiswaViraj) [inline apps/dashboard/src/components/workflow-list.tsx:182]: wrapped it with Link to have url contexts and better UX

Comment (rifont) [reply apps/dashboard/src/components/workflow-list.tsx:182]: 👏 **praise:** Awesome!!‏

**Review by rifont — APPROVED** [review]
🚀

**Pull request merged**